respy/pre_processing/model_processing.py

```diff
@@ -327,59 +327,64 @@ def _parse_types(optim_paras, params):
     implicitly enforced that the parameters form a valid matrix.
 
     """
-    n_choices = len(optim_paras["choices"])
+    if "type_0" in params.index:
+        raise ValueError("Types are numbered from 1 onwards.")
+    if "type_1" in params.index.get_level_values("category"):
+        raise ValueError(
+            "'type_1' cannot be used to specify the probability mass function of types "
+            "as it has to be zero such that all parameters are identified."
+        )
+    if "type_1" in params.index.get_level_values("name"):
+        raise ValueError(
+            "'type_1' cannot be used as a utility covariate as it must be zero due to "
+            "normalization. All other types are expressed in relation to the first."
+        )
+
+    n_types = _infer_number_of_types(params)
 
-    if "type_shift" in params.index:
-        n_types = _infer_number_of_types(params)
-        types = [f"type_{i}" for i in range(2, n_types + 1)]
-        optim_paras["type_covariates"] = (
-            params.loc[types[0]].sort_index().index.to_list()
+    if n_types >= 2:
+        # Parse type probabilities.
+        parsed_parameters = _parse_probabilities_or_logit_coefficients(
+            params, r"\btype_([0-9]+)\b"
         )
+        parsed_parameters = {k - 1: v for k, v in parsed_parameters.items()}
+        default = {i: pd.Series(data=[0], index=["constant"]) for i in range(n_types)}
+        optim_paras["type_prob"] = {**default, **parsed_parameters}
```

Comment:
We had this discussion before and settled on this name. It is not perfect, but I do not want to change it again at this point. But we should bear it in our minds. Maybe even something like ``finite_mixture_distribution``. This could also be a keyword for the parameters in ``params``.